src/raydium/infos.ts

```diff
@@ -472,7 +472,7 @@ infos = class InstanceRaydium {
   static async getFarmerId(farmInfo: FarmInfo, userKey: PublicKey, version: number): Promise<PublicKey> {
     const programId = version === 3 ? FARM_PROGRAM_ID_V3 : FARM_PROGRAM_ID_V5;
 
-    const [farmerId, _] = await PublicKey.findProgramAddress(
+    const [farmerId, _] = await PublicKey.findProgramAddressSync(
```

Comment:
I believe getFarmerId can be a synchronous function here

Reply:
This is a special case here
The `getFarmerId` inside `IInstanceFarm` can't be synchronous.
Its's because in Francium need another `getAssociatedTokenAddress` to derive the farmer PDA.
So all `getFarmerId` need to be async

Reply:
Noted. Let's keep this way for now

Reply:
Correct 
Commit Updated